src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -427,7 +479,7 @@ object RuleUtils {
             baseOutput,
             _,
             _) =>
-        val options = extractBasePath(spark, location)
+        val options = extractBasePath(spark, baseRelation)
```

Comment:
Could you also support DataSourceV2 for this function?

This is necessary for Hybrid Scan - which is going to be enabled by default soon.
Basically, 
- creating the same LogicalRelation as the given source relation, but the list of input files are only appended files.
- union or bucket-wise union (BucketUnion) with this appended data and the index data.

@andrei-ionescu Is it possible to read input parquet files underlying IcebergSource using DataSourceV2 - without `IceBergSource`? if so, we could implement it here. Otherwise, we should define a new source provider API to avoid the lib dependency.

Reply:
👍